benches/src/perfenc.rs

```diff
@@ -119,20 +120,21 @@ impl DisplayUpdates {
 
 #[async_trait::async_trait]
 impl RdpServerDisplayUpdates for DisplayUpdates {
-    async fn next_update(&mut self) -> Option<DisplayUpdate> {
+    async fn next_update(&mut self) -> anyhow::Result<Option<DisplayUpdate>> {
         let stride = self.desktop_size.width as usize * 4;
         let frame_size = stride * self.desktop_size.height as usize;
         let mut buf = vec![0u8; frame_size];
-        if self.file.read_exact(&mut buf).await.is_err() {
-            return None;
-        }
+        // FIXME: AsyncReadExt::read_exact is not cancellation safe.
```

Comment:
The doc comment for `next_update` method says that it must be cancellation-safe. I've added a TODO.